enterprise/glassfish.common/src/org/netbeans/modules/glassfish/common/GlassfishInstance.java

```diff
@@ -201,20 +203,13 @@ public void clear() {
         public String get(Object key) {
             if (GlassfishModule.PASSWORD_ATTR.equals(key)) {
                 String value;
-                synchronized(delegate) {
-                    value = delegate.get(key);
-                }
-                if (value == null) {
-                    value = getPasswordFromKeyring(
-                            delegate.get(GlassfishModule.DISPLAY_NAME_ATTR),
-                            delegate.get(GlassfishModule.USERNAME_ATTR));
-                    synchronized(delegate) {
-                        delegate.put((String)key, value);
-                    }
-                }
+                value = delegate.computeIfAbsent((String)key, k -> 
+                            getPasswordFromKeyring(
+                                delegate.get(GlassfishModule.DISPLAY_NAME_ATTR),
+                                delegate.get(GlassfishModule.USERNAME_ATTR)));
                 return value;
             } else {
-                synchronized(delegate) {return delegate.get(key);}
+                return delegate.computeIfPresent((String)key, (k, v) -> v);
```

Comment:
same here, this is a `get()`

Reply:
The same reason, `synchronized` just a Node instead of the entire map.

Reply:
please elaborate. Why is this here needed and in `java.util.Properties().get()` not?

same question for enterprise/glassfish.common/src/org/netbeans/modules/glassfish/common/GlassfishInstanceProvider.java